### PR TITLE
[CGNS] Add new port and improve hdf5 with features

### LINKED ISSUE #9181: [New Port Request] CGNS
Library name: CGNS

Library description: The CFD General Notation System (CGNS) provides a standard for recording and recovering computer data associated with the numerical solution of fluid dynamics equations.

Source repository URL: https://github.com/CGNS/CGNS

Project homepage (if different from the source repository):

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

```

(a)The path to the HDF5 library should be specified with CMAKE_PREFIX_PATH=$HDF_DIR for linking with a specific HDF5 version.

If HDF5 is built with parallel-IO support via MPI, the 'HDF5_NEEDS_MPI' flag must be set to true.

If HDF5 is built with zlib and szip support, these need to be flagged with 'HDF5_NEEDS_ZLIB' and 'HDF5_NEEDS_SZIP' as well as the paths for those libraries.

(b)Fortran can be enabled by toggling the 'CGNS_ENABLE_FORTRAN' variable.

A view of the attempt to autodetect the correct interface between Fortran and C is show, setting the value of 'FORTRAN_NAMING'.

For gfortran and pgf90 the value of 'FORTRAN_NAMING' shoud be 'LOWERCASE_'.
```

required by #9180

### PR DESCRIPTION
**Pull request by Neumann-A**
closes #9181.

Needs to wait #6939 #9043 merge.

### CODE REVIEW

**Comment by JackBoosY** [comment]
We should test all features in all triplets first.

File: ports/hdf5/CONTROL [hunk 1]
```diff
@@ -1,12 +1,28 @@
 Source: hdf5
-Version: 1.10.5-8
+Version: 1.10.5-9
 Homepage: https://www.hdfgroup.org/downloads/hdf5/
 Description: HDF5 is a data model, library, and file format for storing and managing data
-Build-Depends: zlib, szip
+Build-Depends:
```
Comment (JackBoosY) [inline ports/hdf5/CONTROL]: Could you remove this line?

Thanks.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/hdf5/CONTROL (hunk 1 above)
Comment (Neumann-A) [reply ports/hdf5/CONTROL]: Done.

**Comment by JackBoosY** [comment]
We need to test it in all triplets before merge.

File: ports/cgns/CONTROL [hunk 2]
```diff
@@ -0,0 +1,29 @@
+Source: cgns
+Version: 3.4.0
+Homepage: http://cgns.org/
+Description: The CFD General Notation System (CGNS) provides a standard for recording and recovering computer data associated with the numerical solution of fluid dynamics equations. 
+Build-Depends: 
+Default-Features: hdf5, lfs
+
+Feature: mpi
+Description: Compile with support for the message passage interface (MPI)
+Build-Depends: hdf5[mpi]
```
Comment (JackBoosY) [inline ports/cgns/CONTROL]: This feature depends on `hdf5[mpi]`, but `hdf5[mpi]` does not exist. Should it be `hdf5[parallel]`?

**Review by JackBoosY — CHANGES_REQUESTED** [review]
```
-- Performing post-build validation
The following EXEs were found in /bin or /debug/bin. EXEs are not valid distribution targets.

    F:/9183/vcpkg/packages/hdf5_x86-windows/bin/ph5diff.exe

The following EXEs were found in /bin or /debug/bin. EXEs are not valid distribution targets.

    F:/9183/vcpkg/packages/hdf5_x86-windows/debug/bin/ph5diff.exe

Found 2 error(s). Please correct the portfile:
    F:\9183\vcpkg\ports\hdf5\portfile.cmake
-- Performing post-build validation done
Error: Building package hdf5:x86-windows failed with: POST_BUILD_CHECKS_FAILED
```
Please fix this issue.

File: ports/cgns/CONTROL (hunk 2 above)
Comment (Neumann-A) [reply ports/cgns/CONTROL]: good catch. I have corrected it.

**Comment by JackBoosY** [comment]
Build cgns using command `./vcpkg install cgns[fortran]` cause error in configuration steps:
```
CMake Error at CMakeLists.txt:117 (enable_language):
  The Ninja generator does not support Fortran using Ninja version

    1.8.2

  due to lack of required features.  Kitware has implemented the required
  features but as of this version of CMake they have not been integrated to
  upstream ninja.  Pending integration, Kitware maintains a branch at:

    https://github.com/Kitware/ninja/tree/features-for-fortran#readme

  with the required features.  One may build ninja from that branch to get
  support for Fortran.


-- Configuring incomplete, errors occurred!
```

File: ports/cgns/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+)
+
+vcpkg_install_cmake()
+
+file(INSTALL ${CURRENT_PACKAGES_DIR}/include/cgnsBuild.defs DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
+file(REMOVE ${CURRENT_PACKAGES_DIR}/include/cgnsBuild.defs ${CURRENT_PACKAGES_DIR}/include/cgnsconfig.h)
+
+file(INSTALL ${CURRENT_PORT_DIR}/cgnsconfig.h DESTINATION ${CURRENT_PACKAGES_DIR}/include) # we patched the config and the include is all that is needed
+
+IF(EXISTS ${CURRENT_PACKAGES_DIR}/debug)
```
Comment (JackBoosY) [inline ports/cgns/portfile.cmake]: Can we use macros to handle this part of the code? I want to reduce this part of the code.

File: ports/hdf5/portfile.cmake [hunk 4]
```diff
@@ -17,49 +17,136 @@ vcpkg_extract_source_archive_ex(
...
+                ${CURRENT_PACKAGES_DIR}/debug/bin/h5copy-shared${VCPKG_TARGET_EXECUTABLE_SUFFIX}
+                ${CURRENT_PACKAGES_DIR}/debug/bin/h5diff-shared${VCPKG_TARGET_EXECUTABLE_SUFFIX}
+                ${CURRENT_PACKAGES_DIR}/debug/bin/h5dump-shared${VCPKG_TARGET_EXECUTABLE_SUFFIX}
+                ${CURRENT_PACKAGES_DIR}/debug/bin/h5ls-shared${VCPKG_TARGET_EXECUTABLE_SUFFIX}
+                ${CURRENT_PACKAGES_DIR}/debug/bin/h5repack-shared${VCPKG_TARGET_EXECUTABLE_SUFFIX}
+                ${CURRENT_PACKAGES_DIR}/debug/bin/h5stat-shared${VCPKG_TARGET_EXECUTABLE_SUFFIX})
+        else()
+            set(DEBUG_SHARED_BINS )
+        endif()
+        file(REMOVE ${CURRENT_PACKAGES_DIR}/debug/bin/gif2h5${VCPKG_TARGET_EXECUTABLE_SUFFIX}
```
Comment (JackBoosY) [inline ports/hdf5/portfile.cmake]: Can we use macros to handle this part of the code? I want to reduce this part of the code.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by Neumann-A** [comment]
>Please fix this issue.

will do so if your single build PR for HDF5 is merged into master. (#9043) 
This PR is basically waiting for it to so that i can rewrite the tool installation/movement. 

> Build cgns using command ./vcpkg install cgns[fortran] cause error in configuration steps:

Please read the feature description.
```
Feature: fortran
Description: Enable fortran support (not yet implemented) 
```
Fortran is not yet merged into VCPKG. Needs #6939.

File: ports/cgns/portfile.cmake (hunk 3 above)
Comment (Neumann-A) [reply ports/cgns/portfile.cmake]: that is the plan after PR 9043 is merged. I'll use the same approach as in e.g. VTK

**Comment by Neumann-A** [comment]
@JackBoosY: Do you mind if I close this PR and let it be superseded by #9202? (because I would need to backport some changes from #9202 to this PR)

**Comment by JackBoosY** [comment]
@Neumann-A Okay.

File: ports/cgns/portfile.cmake (hunk 3 above)
Comment (Neumann-A) [reply ports/cgns/portfile.cmake]: done in https://github.com/microsoft/vcpkg/pull/9202/commits/3595c5aeb9cd8a67a930cb0614a8fe2fd2181e2d

File: ports/hdf5/portfile.cmake (hunk 4 above)
Comment (Neumann-A) [reply ports/hdf5/portfile.cmake]: done in https://github.com/microsoft/vcpkg/pull/9202/commits/3595c5aeb9cd8a67a930cb0614a8fe2fd2181e2d

**Pull request closed without merging**